src/main/java/org/disq_bio/disq/impl/file/HadoopFileSystemWrapper.java

```diff
@@ -55,57 +56,67 @@ public boolean usesNio() {
   @Override
   public String normalize(Configuration conf, String path) throws IOException {
     Path p = new Path(path);
-    FileSystem fileSystem = p.getFileSystem(conf);
+    FileSystem fileSystem = getFileSystem(conf, p);
     return fileSystem.makeQualified(p).toString();
   }
 
+  private static FileSystem getFileSystem(final Configuration conf, final Path p)
+      throws IOException {
+    final FileSystem fileSystem = p.getFileSystem(conf);
+    if (fileSystem instanceof LocalFileSystem) {
+      return ((LocalFileSystem) fileSystem).getRawFileSystem();
```

Comment:
Add a comment explaining this special casing of `LocalFileSystem` (or, if we can't explain it, at least provide a comment with a reference for where the fix came from).